src/handlers/buyselltrade.ts

```diff
@@ -0,0 +1,138 @@
+import { Message, Client, TextChannel, MessageEmbed } from "discord.js";
+import { clearTimeout } from "timers";
+import config from '../config';
+
+class BstPost extends Message {
+    validationMessages: string[];
+    // folowing only set when validation fails 
+    warningMessage?: Message;
+    expiration?: NodeJS.Timeout;
+
+    constructor(msg: Message, warningMsg?: Message, expiration?: NodeJS.Timeout) {
+        super(msg.client, {
+            id: msg.id,
+            type: msg.type,
+            content: msg.content,
+            author: msg.author,
+            pinned: msg.pinned,
+            tts: msg.tts,
+            embeds: msg.embeds,
+            attachments: msg.attachments,
+            nonce: msg.nonce
+        }
+            , msg.channel);
+
+        this.validationMessages = [];
+        this.warningMessage = warningMsg;
+        this.expiration = expiration;
+    }
+    addMessage(errText: string) {
+        this.validationMessages.push(errText);
+    }
+    setMessageExpiration(timeInMs: number) {
+        setTimeout(this.deleteMessage.bind(this), timeInMs);
+    }
+    async deleteMessage() {
+        await this.delete();
+        this.deleted = true;
+        await this.warningMessage?.edit(this.warningMessage.content + "**Post was not edited in 30 minutes and was deleted**");
+    }
+    clearMessageExpiration() {
+        if (this.expiration)
+            clearTimeout(this.expiration);
+    }
+}
+
+export default async function handleBuySellTradeMessage(
+    msg: Message,
+    client: Client
+): Promise<void> {
+    if (msg.channel.id == config.FORTIES_BST) {
+        const currPost = new BstPost(msg);
+        validateBstMessage(currPost);
+    }
+}
+
+async function validateBstMessage(bstPost: BstPost) {
+    // add any additional validations here 
+    if (validateLocationBlockExistance(bstPost)) {
+        validateLocationBlockLength(bstPost);
+    }
+
+    if (bstPost.validationMessages.length) {
+        await handleFailedBstValidation(bstPost);
+    }
+    if (bstPost.warningMessage) {
+        // if this isn't a messages first time through the system, clear the delete timer
+        bstPost.clearMessageExpiration();
+    }
+}
+
+async function handleFailedBstValidation(bstPost: BstPost) {
+    const channel = bstPost.guild?.channels.cache.get(config.FORTIES_BST_DISCUSSION);
+    // this is wonky for sure - discordJS returns a type "GuildChannel" on  
+    // lookup so we have to cast to a TextChannel to send any messages, see : 
+    // https://stackoverflow.com/questions/53563862/send-message-to-specific-channel-with-typescript/53565548
+    if (!((channel): channel is TextChannel => channel?.type === 'text')(channel)) return;
+
+    bstPost.warningMessage = await channel.send(createFailMessage(bstPost));
+
+    runDeleteCountDown(bstPost);
+}
+
+function runDeleteCountDown(bstPost: BstPost) {
```

Comment:
The way this system of multiple interacting timers works is a little bit hard to follow because some of it is using side-effects on the `BstPost` object, and other parts are being handled in module-level functions.

I think it would be way easier to understand and also Type correctly if you could use just one module-level function to manage all of it.